### PR TITLE
Incorporate Steven's comments

### LINKED ISSUE #1: Tag a release
@stevengj Do you have any input on what should be done here before this is tagged and registered?

### PR DESCRIPTION
**Pull request by ararslan**
I'll put all of the requested changes into this PR so that it can be tagged once this is merged.

Fixes #1

### CODE REVIEW

File: src/QuadGK.jl [hunk 1]
```diff
@@ -318,10 +318,14 @@ function eigvec1(b,z::Number,m=length(b)+1)
...
 
-# return (x, w) pair of N quadrature points x[i] and weights w[i] to
-# integrate functions on the interval (-1, 1).  i.e. dot(f(x), w)
-# approximates the integral.  Uses the method described in Trefethen &
-# Bau, Numerical Linear Algebra, to find the N-point Gaussian quadrature
+"""
+    gauss([T,] N)
+
+Return a pair `(x, w)` of `N` quadrature points `x[i]` and weights `w[i]` to
+integrate functions on the interval `(-1, 1)`,  i.e. `dot(f(x), w)`
```
Comment (stevengj) [inline src/QuadGK.jl]: `dot(w, f.(x))`, so that you don't conjugate `f`.  Or `sum(w .* f.(x))` is probably more natural here.

Comment (stevengj) [inline src/QuadGK.jl]:
```
`T` is an optional parameter specifying the floating-point type, defaulting to `Float64`.
Arbitrary precision (`BigFloat`) is also supported.
```

File: src/QuadGK.jl [hunk 2]
```diff
@@ -318,10 +318,14 @@ function eigvec1(b,z::Number,m=length(b)+1)
...
+approximates the integral.  Uses the method described in Trefethen &
+Bau, Numerical Linear Algebra, to find the `N`-point Gaussian quadrature.
```
Comment (stevengj) [inline src/QuadGK.jl]: ... in O(`N`²) operations

File: src/QuadGK.jl [hunk 3]
```diff
@@ -333,11 +337,17 @@ function gauss{T<:AbstractFloat}(::Type{T}, N::Integer)
...
+
+"""
+    kronrod([T,] n)
+
+Compute `2n+1` Kronrod points `x` and weights `w` based on the description in
+Laurie (1997), appendix A, simplified for `a=0`, for integrating on `[-1,1]`.
+Since the rule is symmetric, this only returns the `n+1` points with `x <= 0`.
+The function Also computes the embedded `n`-point Gauss quadrature weights `gw`
+(again for `x <= 0`), corresponding to the points `x[2:2:end]`. Returns `(x,w,wg)`.
+"""
```
Comment (stevengj) [inline src/QuadGK.jl]:
```
Given these points and weights, the estimated integral `I` and error `E` can be computed for an
integrand `f(x)` as follows:

    x,w,wg = kronrod(n)
    fx⁰ = f(x[end])      # f(0)
    x⁻ = x[1:end-1]     # the x < 0 Kronrod points
    fx = f.(x⁻) .+ f.((-).(x⁻))    # f(x < 0) + f(x > 0)
    I = sum(fx .* w[1:end-1]) + fx⁰ * w[end]
    if isodd(n)
        E = abs(sum(fx[2:2:end] .* wg[1:end-1]) + fx⁰*wg[end] - I)
    else
        E = abs(sum(fx[2:2:end] .* wg[1:end])- I)
    end
```

Also, should have a similar comment to above explaining `T` and noting that it is O(N^2).

**Comment by stevengj** [comment]
See also my suggestion on the `kronrod` docstring.

**Comment by ararslan** [comment]
How are things looking now?

**Comment by stevengj** [comment]
Shouldn't `docs/src/index.md` be updated to pull in the new docstrings?

**Comment by ararslan** [comment]
Good call, thanks!

**Comment by stevengj** [comment]
LGTM

**Comment by ararslan** [comment]
Fantastic. Thanks for all your help!

**Pull request merged**